ccl/kvccl/kvfollowerreadsccl: TestBoundedStalenessDataDriven failed

ccl/kvccl/kvfollowerreadsccl.TestBoundedStalenessDataDriven [failed](https://tanzanite.cluster.engflow.com/invocations/default/2364f647-7f56-4649-a42d-69307fbef623?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL2t2Y2NsL2t2Zm9sbG93ZXJyZWFkc2NjbDprdmZvbGxvd2VycmVhZHNjY2xfdGVzdA==) on master @ [47699f3887ad5d1b8c7c5905eb5c49628aa59bbe](https://github.com/cockroachdb/cockroach/commits/47699f3887ad5d1b8c7c5905eb5c49628aa59bbe):


```
         * event 1: colbatchscan trace on node_idx 2: local follower read
    datadriven.go:144: 
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/kvccl/kvfollowerreadsccl/kvfollowerreadsccl_test_/kvfollowerreadsccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/kvccl/kvfollowerreadsccl/testdata/boundedstaleness/single_row:125:
        override-matching-stmt-for-tracing [0 args]
        SELECT pk FROM t AS OF SYSTEM TIME with_min_timestamp(now() - '10s') WHERE pk = 1
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/kvccl/kvfollowerreadsccl/kvfollowerreadsccl_test_/kvfollowerreadsccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/kvccl/kvfollowerreadsccl/testdata/boundedstaleness/single_row:129:
        query [1 args]
        EXECUTE min_timestamp_prep
        ----
        1
        events (1 found):
         * event 1: colbatchscan trace on node_idx 2: local follower read
    datadriven.go:144: 
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/kvccl/kvfollowerreadsccl/kvfollowerreadsccl_test_/kvfollowerreadsccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/kvccl/kvfollowerreadsccl/testdata/boundedstaleness/single_row:136:
        reset-matching-stmt-for-tracing [0 args]
        <no input to command>
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/kvccl/kvfollowerreadsccl/kvfollowerreadsccl_test_/kvfollowerreadsccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/kvccl/kvfollowerreadsccl/testdata/boundedstaleness/single_row:141:
        exec [0 args]
        SET CLUSTER SETTING kv.closed_timestamp.target_duration = '1hr';
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/kvccl/kvfollowerreadsccl/kvfollowerreadsccl_test_/kvfollowerreadsccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/kvccl/kvfollowerreadsccl/testdata/boundedstaleness/single_row:145:
        exec [0 args]
        ALTER TABLE t ADD COLUMN new_col INT NOT NULL DEFAULT 2
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/kvccl/kvfollowerreadsccl/kvfollowerreadsccl_test_/kvfollowerreadsccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/kvccl/kvfollowerreadsccl/testdata/boundedstaleness/single_row:151:
        query [1 args]
        SELECT * FROM t AS OF SYSTEM TIME with_max_staleness('10s') WHERE pk = 1
        ----
        1 2
        events (1 found):
         * event 1: colbatchscan trace on node_idx 2: local read then remote leaseholder read
    datadriven.go:144: 
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/kvccl/kvfollowerreadsccl/kvfollowerreadsccl_test_/kvfollowerreadsccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/kvccl/kvfollowerreadsccl/testdata/boundedstaleness/single_row:158:
         SELECT * FROM t AS OF SYSTEM TIME with_min_timestamp(now() - '10s') WHERE pk = 1
        expected:
        1 2
        events (1 found):
         * event 1: colbatchscan trace on node_idx 2: local read then remote leaseholder read
        
        found:
        1 2
        events (1 found):
         * event 1: colbatchscan trace on node_idx 2: local read
    --- FAIL: TestBoundedStalenessDataDriven/single_row (28.69s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBoundedStalenessDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45906

ccl/kvccl/kvfollowerreadsccl.TestBoundedStalenessDataDriven [failed](https://tanzanite.cluster.engflow.com/invocations/default/c10b4ef4-6c60-44f5-83a7-801c55cd8657?testReportRun=15&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL2t2Y2NsL2t2Zm9sbG93ZXJyZWFkc2NjbDprdmZvbGxvd2VycmVhZHNjY2xfdGVzdA==) on master @ [097438ac38e411b0fde101ebcae5cf97a798d1db](https://github.com/cockroachdb/cockroach/commits/097438ac38e411b0fde101ebcae5cf97a798d1db):


```
=== RUN   TestBoundedStalenessDataDriven
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestBoundedStalenessDataDriven1729361669
    test_log_scope.go:76: use -show-logs to present logs inline
    boundedstaleness_test.go:422: -- test log scope end --
--- FAIL: TestBoundedStalenessDataDriven (11.62s)
=== RUN   TestBoundedStalenessDataDriven/single_row
    datadriven.go:144: 
        /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/kvccl/kvfollowerreadsccl/kvfollowerreadsccl_test_/kvfollowerreadsccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/kvccl/kvfollowerreadsccl/testdata/boundedstaleness/single_row:5:
        exec [0 args]
        SET CLUSTER SETTING kv.closed_timestamp.target_duration = '1s';
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/kvccl/kvfollowerreadsccl/kvfollowerreadsccl_test_/kvfollowerreadsccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/kvccl/kvfollowerreadsccl/testdata/boundedstaleness/single_row:9:
        exec [0 args]
        CREATE TABLE t(pk INT PRIMARY KEY);
        INSERT INTO t VALUES (1);
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/kvccl/kvfollowerreadsccl/kvfollowerreadsccl_test_/kvfollowerreadsccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/kvccl/kvfollowerreadsccl/testdata/boundedstaleness/single_row:18:
        query [2 args]
        SELECT * FROM t AS OF SYSTEM TIME with_max_staleness('1μs') WHERE pk = 1
        ----
        1
        events (1 found):
         * event 1: colbatchscan trace on node_idx 2: local read then remote leaseholder read
    datadriven.go:144: 
        /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/kvccl/kvfollowerreadsccl/kvfollowerreadsccl_test_/kvfollowerreadsccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/kvccl/kvfollowerreadsccl/testdata/boundedstaleness/single_row:25:
         SELECT * FROM t AS OF SYSTEM TIME with_min_timestamp(now() - '1μs') WHERE pk = 1
        expected:
        1
        events (1 found):
         * event 1: colbatchscan trace on node_idx 2: local read then remote leaseholder read
        
        found:
        1
        events (1 found):
         * event 1: colbatchscan trace on node_idx 2: local read
    --- FAIL: TestBoundedStalenessDataDriven/single_row (9.05s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=15</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBoundedStalenessDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/kvccl/kvfollowerreadsccl/BUILD.bazel
+++ b/pkg/ccl/kvccl/kvfollowerreadsccl/BUILD.bazel
@@ -44,8 +44,7 @@ go_test(
     data = glob(["testdata/**"]),
     embed = [":kvfollowerreadsccl"],
     exec_properties = select({
-        "//build/toolchains:is_heavy": {"test.Pool": "large"},
-        "//conditions:default": {"test.Pool": "default"},
+        "//conditions:default": {"test.Pool": "large"},
     }),
     deps = [
         "//pkg/base",

--- a/pkg/ccl/kvccl/kvfollowerreadsccl/boundedstaleness_test.go
+++ b/pkg/ccl/kvccl/kvfollowerreadsccl/boundedstaleness_test.go
@@ -264,6 +264,7 @@ func TestBoundedStalenessDataDriven(t *testing.T) {
 	const msg = "1μs staleness reads may actually succeed due to the slow environment"
 	skip.UnderStress(t, msg)
 	skip.UnderRace(t, msg)
+	skip.UnderDeadlock(t, msg)
 	defer ccl.TestingEnableEnterprise()()
 
 	ctx := context.Background()

